### DISCUSSION TITLE
Picamera2 AfWindows

### DISCUSSION THREAD

**Discussion by waikeat2397** [raspberrypi/picamera2#1038] [General]
Hi, I am trying on picamera2 AfWindows feature. Found out something weird. 
I am using Rapberry Pi Camera Module 3. 
Set the sensor mode at (4608, 2592). Trying to set the AfWindows accordingly, it is working fine!

Now I set the sensor mode at (2304, 1296), Trying to set the AfWindows accordingly to the resolution, found out that it is not working as expected. I need to set the AfWindows x and y offset according to full resolution sensor mode. Is this valid?

Below is my sample code. 

```from libcamera import controls
from picamera2 import Picamera2
import time

picam2 = Picamera2()
camera_config = picam2.create_still_configuration(main={"size": (2304, 1296), 'format': 'BGR888'}, raw={"size": (2304, 1296)})
picam2.configure(camera_config)
picam2.start()
time.sleep(1)
picam2.set_controls({'AeEnable': True})
time.sleep(3)
picam2.set_controls({'AeEnable': False})
#823, 1770, 1028, 517
picam2.set_controls({'AfMetering': controls.AfMeteringEnum.Windows, 'AfWindows': [(823, 1770, 1028, 517)], 'AfMode': controls.AfModeEnum.Auto})
picam2.autofocus_cycle()
print(picam2.capture_metadata()['LensPosition'])
picam2.capture_file('test.jpg')
#1851, 853, 647, 304
picam2.set_controls({'AfMetering': controls.AfMeteringEnum.Windows, 'AfWindows': [(1851, 853, 647, 304)], 'AfMode': controls.AfModeEnum.Auto})
picam2.autofocus_cycle()
print(picam2.capture_metadata()['LensPosition'])
picam2.capture_file('test2.jpg')


picam2.stop()
picam2.close()
```

**Comment by davidplowman** [comment]
I think the AF windows are always given in the coordinates of the full sensor resolution. The idea is that you get the same AF behaviour regardless of what mode the camera is actually in. Does this accord with your findings?
